src/libs/highlevel/README.md

```diff
@@ -0,0 +1,302 @@
+either use the function `int elektraGetEnumInt (Elektra * elektra, char * keyName)`, and deal with the raw integer yourself, or use the 
+convenience macro `elektraGetEnum(elektra, keyname, enumType)`, which calls `elektraGetEnumInt` and then casts to `enumType`.
+
+```c
+typedef enum { A, B, C } MyEnum;
+
+Elektra * elektra = elektraOpen ("/sw/org/myapp/#0/current", NULL, NULL);
+
+// Read raw int value
+int value = elektraGetEnumInt (elektra, "message");
+MyEnum enumValue = elektraGetEnum(elektra, "message", MyEnum);
+
+// enumValue == (MyEnum) value
+
+elektraClose (elektra);
+```
+
+To write an enum value use `void elektraSetEnumInt (Elektra * elektra, char * name, int value, ElektraError ** error)`. A convenience
+macro is not provided, because you can simply pass the enum value as an `int` argument.
+
+Similar functions and macros are provided for array elements:
+
+```c
+int elektraGetEnumIntArrayElement (Elektra * elektra, char * keyName, size_t index)
+
+elektraGetEnumArrayElement(elektra, keyname, index, enumType)
+
+void elektraSetEnumIntArrayElement (Elektra * elektra, char * name, size_t index, int value, ElektraError ** error)
+```
+
+### Raw Values
+You can use `const char * elektraGetValue (Elektra * elektra, const char * name)` to read the raw (string) value of a key. No type checking
+or type conversion will be attempted. Additionally this function is guaranteed to not call the fatal error handler. It will simply return
+`NULL`, if the key was not found.
+
+If you want to set a raw value (e.g. if you want to extend the API with your own custom types), use 
+`void elektraSetValue (Elektra * elektra, const char * name, const char * value, KDBType type, ElektraError ** error)`. Obviously you have
+to provide a type for the value you set, so that the API can perform type checking, when reading the value next time.
+
+Similar functions are provided for array elements:
+
+```c
+const char * elektraGetArrayElementValue (Elektra * elektra, const char * name, size_t index)
+
+void elektraSetArrayElementValue (Elektra * elektra, const char * name, size_t index, const char * value, KDBType type, ElektraError ** error)
+```
+
+#### Type Information
+The type information is stored in the `"type"` meta-key. `KDBType elektraGetType (Elektra * elektra, const char * keyname)` (or
+`KDBType elektraGetArrayElementType (Elektra * elektra, const char * name, size_t index)` for array elements) lets you access this
+information. A setter is not provided, because changing the type of a key without changing its value rarely, if ever, makes sense.
+
+### Binary Values
+The high-level API does not support binary key values at this time.
+
+## Example
+
+```c
+#include <stdio.h>
+#include <elektra.h>
+
+int main ()
+{
+  ElektraError * error = NULL;
+  Elektra * elektra = elektraOpen ("/sw/org/myapp/#0/current", NULL, &error);
+
+  if (error != NULL) 
```

Comment:
`Elektra` is only allocated, if no error occured.

Reply:
Yes, so I would expect the error handling to be done using elektra == NULL?

Because checking for the presence of the "error" pointer might be wrong if there are warnings without errors? Or how is this handled?